src/kwp2000/read_dtc_by_status.rs

```diff
@@ -79,16 +79,16 @@ impl Kwp2000DiagnosticServer {
     }
 
     /// Returns a list of all supported DTCs on the ECU regardless of their status in ISO15031-6 format
-    pub fn read_supported_dtcs_iso15031(&mut self, range: DTCRange) -> DiagServerResult<Vec<DTC>> {
+    pub fn kwp_read_supported_dtcs_iso15031(&mut self, range: DTCRange) -> DiagServerResult<Vec<DTC>> {
         let res: Vec<DTC> = Vec::new();
 
         loop {
-            let res_bytes = self.execute_command_with_response(
+            let res_bytes = self.send_command_with_response(
                 KWP2000Command::ReadDiagnosticTroubleCodesByStatus,
                 &range.as_args(0x01),
             )?;
             println!("RES: {:02X?}", res_bytes);
-            match self.read_extended_supported_dtcs(range) {
+            match self.kwp_read_extended_supported_dtcs(range) {
```

Comment:
just an idea here, don't have to change it if you don't want to. Match guards tend to be cleaner in some cases:

```suggestion
            match self.kwp_read_extended_supported_dtcs(range) {
                Ok(x) if x == 0 => break, // Completed reading
                Err(_) => break, // Return what we have
                _ => {} // Else keep looping to read DTCs
            }
```

Reply:
Dope, I realized you don't even need the match guard here, you can just use `Ok(0)` as a match statement

Reply:
i think this will make code easier to read